src/main/java/org/sagebionetworks/bridge/udd/dynamodb/DynamoHelper.java

```diff
@@ -57,44 +86,79 @@ public StudyInfo getStudy(String studyId) {
 
-        // convert healthId to healthCode
-        Item healthIdItem = ddbHealthIdTable.getItem("id", accountInfo.getHealthId());
-        String healthCode = healthIdItem.getString("code");
-
-        // get uploads from healthCode-uploadDate-index
-        Iterable<Item> uploadIter = queryHelper("healthCode", healthCode,
-                new RangeKeyCondition("uploadDate").between(startDateString, endDateString));
-
-        List<UploadInfo> uploadInfoList = new ArrayList<>();
-        for (Item oneUpload : uploadIter) {
-            String uploadId = oneUpload.getString("uploadId");
-            String uploadDateStr = oneUpload.getString("uploadDate");
-            UploadInfo oneUploadInfo = new UploadInfo.Builder().withId(uploadId)
-                    .withUploadDate(uploadDateStr).build();
-            uploadInfoList.add(oneUploadInfo);
+            UploadSchema schema = UploadSchema.fromDdbItem(fullSchemaItem);
+            schemaList.add(schema);
+        }
+
+        // Now query the SynapseTables table to get the Synapse table IDs for the schema. We use a reverse map from
+        // Synapse table ID to upload schema, because multiple upload schemas can map to a single Synapse table. (This
+        // is due to some early day hacks in the original studies.)
+        Multimap<String, UploadSchema> synapseToSchemaMultimap = HashMultimap.create();
+        for (UploadSchema oneSchema : schemaList) {
+            Item synapseMapRecord = ddbSynapseMapTable.getItem("schemaKey", oneSchema.getKey().toString());
+            if (synapseMapRecord == null) {
+                // This could happen if the schema was just created, but the Bridge-Exporter hasn't created the
+                // corresponding Synapse table yet. If so, there's obviously no data. Skip this one.
+                continue;
+            }
+
+            String synapseTableId = synapseMapRecord.getString("tableId");
+            synapseToSchemaMultimap.put(synapseTableId, oneSchema);
         }
-        return uploadInfoList;
+
+        // Dedupe the upload schemas. We pick the canonical schema based on which one has the highest rev.
+        Map<String, UploadSchema> synapseToSchemaMap = new HashMap<>();
+        for (String oneSynapseTableId : synapseToSchemaMultimap.keySet()) {
+            Iterable<UploadSchema> schemaIter = synapseToSchemaMultimap.get(oneSynapseTableId);
+            UploadSchema canonicalSchema = null;
+            for (UploadSchema oneSchema : schemaIter) {
+                if (canonicalSchema == null ||
+                        canonicalSchema.getKey().getRevision() < oneSchema.getKey().getRevision()) {
+                    canonicalSchema = oneSchema;
+                }
+            }
+
+            // Because of the way this code is written, there will always be at least one schema for this table ID, so
+            // by this point, canonicalSchema won't be null.
+            synapseToSchemaMap.put(oneSynapseTableId, canonicalSchema);
+        }
```

Comment:
To clarify, in some obscure edge cases, multiple schemas point at the same Synapse table. In this case, querying against just that one Synapse table is sufficient. In this case, the schemas are always compatible.

This is no longer possible going forward (unless we manually hack up the Synapse table mapping). The "canonical schema" is only needed her for backwards compatibility.